src/tests/unit/ingredients-dao.test.js

```diff
@@ -239,4 +247,60 @@ describe('test ingredients dao', () => {
       });
     });
   });
+  context('getIngredientById', () => {
+    let result;
+    let id;
+    beforeEach(() => {
+      ingredientsDao = proxyquireIngredientsDao(
+        connectionStub,
+        serializeIngredientsStub,
+        serializeIngredientStub,
+      );
+      result = ingredientsDao.getIngredientById(id);
+    });
+
+    afterEach(() => {
+      connectionStub.resetHistory();
+      serializeIngredientStub.resetHistory();
+      serializeIngredientStub.resetHistory();
+    });
+
+    context('when it gets an integer-formatted id', () => {
+      before(() => {
+        connectionStub = sinon.stub().returns(singleRecordReturn);
+        id = '1';
+      });
```

Comment:
This is a stab at actually solving the DRY problem by initializing stubs and such in a `before` hook while assigning the result in a top level `beforeEach` hook. Replicates the (imo better) style I've used in [ginkgo](https://onsi.github.io/ginkgo/#separating-creation-and-configuration-justbeforeeach) for BDD tests that uses `beforeEach` and `justBeforeEach`. Then `context` blocks can reassign stubs that are used specifically for that context, such as changing the database return, without having to duplicate logic, and we can minimize the amount of logic in each `it` block.

Reply:
Nice work! Looks real good.